docs/docs/APIdocs/layers/nn/Convolution Layers/SpatialConvolution.md

```diff
@@ -0,0 +1,157 @@
+## SpatialConvolution ##
+
+**Scala:**
+```scala
+val m = SpatialConvolution[T](nInputPlane,nOutputPlane,kernelW,kernelH,strideW,strideH,padW,padH,nGroup,propagateBack,wRegularizer,bRegularizer,initWeight, initBias, initGradWeight, initGradBias)
+```
+**Python:**
+```python
+m = SpatialConvolution(n_input_plane,n_output_plane,kernel_w,kernel_h,stride_w=1,stride_h=1,pad_w=0,pad_h=0,n_group=1,propagate_back=True,init_method="default")
+```
+
+SpatialConvolution is a module that applies a 2D convolution over an input image.
+
+The input tensor in `forward(input)` is expected to be
+either a 4D tensor (`batch x nInputPlane x height x width`) or a 3D tensor (`batch x height x width`). The convolution is performed on the last two dimensions.
+
+**Scala example:**
+```scala
+scala> val input = Tensor[Double](1,2,3,3).randn()
+input: com.intel.analytics.bigdl.tensor.Tensor[Double] =
+(1,1,.,.) =
+-0.15036706177525752    0.6812241325457827      -0.6850390156337345
+-0.1914930670115203     -0.5915171727695424     -0.6514810516405148
+-0.6068519920134171     0.16450267576382982     -0.2486342371203115
+
+(1,2,.,.) =
+-0.9283595025163413     -0.02194934330100707    0.7781211906256997
+-1.100526902178938      -0.5722917467649458     1.0299206067029267
+1.2994160475191487      -0.5502152506035778     2.0080993032094896
+
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 1x2x3x3]
+
+scala> val m = SpatialConvolution[Double](2,1,2,2,1,1,0,0)
+m: com.intel.analytics.bigdl.nn.SpatialConvolution[Double] = SpatialConvolution[9a2fb54a](2 -> 1, 2 x 2, 1, 1, 0, 0)
+
+scala> m.setInitMethod(weightInitMethod = BilinearFiller, biasInitMethod = Zeros)
+res41: m.type = SpatialConvolution[9a2fb54a](2 -> 1, 2 x 2, 1, 1, 0, 0)
+
+scala> m.getParameters()
+res43: (com.intel.analytics.bigdl.tensor.Tensor[Double], com.intel.analytics.bigdl.tensor.Tensor[Double]) =
+(1.0
+0.0
+0.0
+0.0
+1.0
+0.0
+0.0
+0.0
+0.0
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 9],0.0
+0.0
+0.0
+0.0
+0.0
+0.0
+0.0
+0.0
+0.0
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 9])
+
+scala> m.forward(input)
+res44: com.intel.analytics.bigdl.tensor.Tensor[Double] =
+(1,1,.,.) =
+-1.0787265642915989     0.6592747892447757
+-1.2920199691904581     -1.163808919534488
+
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 1x1x2x2]
+
+scala> val gradOut = Tensor[Double](1,1,2,2).fill(0.2)
+gradOut: com.intel.analytics.bigdl.tensor.Tensor[Double] =
+(1,1,.,.) =
+0.2     0.2
+0.2     0.2
+
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 1x1x2x2]
+
+scala> m.backward(input,gradOut)
+res45: com.intel.analytics.bigdl.tensor.Tensor[Double] =
```

Comment:
changed in the new pull request #1122. BTW, as I understand,  we should at least put forward in the example, backward is not necessary but no need to remove if you added any. And we're free to add anything we think user should know. Differences in styles and contents can be tolerated as long as the doc is clear.